src/Summoner/Project.hs

```diff
@@ -54,7 +54,7 @@ generateProject projectName Config{..} = do
     settingsGithub <- decisionToBool cGitHub "GitHub integration"
     settingsTravis <- ifGithub settingsGithub "Travis CI integration" cTravis
     settingsAppVey <- ifGithub (settingsStack && settingsGithub) "AppVeyor CI integration" cAppVey
-    settingsPrivat <- ifGithub settingsGithub "private repository" cPrivate
+    settingsPrivat <- ifGithub settingsGithub "private repository" $ if noUpload then Nop else cPrivate
```

Comment:
I guess it can be written a bit easier:
```suggestion
    settingsPrivat <- ifGithub (settingsGithub && not noUpload) "private repository" cPrivate
```
Though I didn't checked it manually.